### PR TITLE
NETOBSERV-1756 Allow filtering based on TCPFlags values

### PR DESCRIPTION
**Pull request by msherif1234**
## Description

Allow filtering based on TCP flags

## Dependencies
https://github.com/netobserv/netobserv-ebpf-agent/pull/367
https://github.com/netobserv/network-observability-operator/pull/710

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Review by jpinsonneau — APPROVED** [review]
Thanks @msherif1234 !

**Comment by msherif1234** [comment]
image

**Review by jpinsonneau — APPROVED** [review]
Thanks @msherif1234 !

**Review by jpinsonneau — APPROVED** [review]
Thanks for adding the display part ! LGTM

File: web/src/components/drawer/record/record-field.tsx [hunk 1]
```diff
@@ -470,6 +475,22 @@ export const RecordField: React.FC<RecordFieldProps> = ({
         }
         return singleContainer(child);
       }
+      case ColumnsId.tcpflags: {
+        let child = emptyText();
+        if (typeof value === 'number' && !isNaN(value)) {
+          const serviceClassName = gettcpFlagsServiceClassName(value);
```
Comment (jotak) [inline web/src/components/drawer/record/record-field.tsx:481]: If I read correctly, this wouldn't decompose the OR-combination, e.g. we can have 3 that is 1+2 = FIN+SYN

Comment (msherif1234) [reply web/src/components/drawer/record/record-field.tsx:481]: ebpf agent doesn't generate this combo here is what we expect to see in the flags field today https://github.com/netobserv/netobserv-ebpf-agent/blob/main/bpf/utils.h#L11

Comment (msherif1234) [reply web/src/components/drawer/record/record-field.tsx:481]: I updated the PR to bring in more combos but not the entire list as there some flags with less interest to decode but if I am wrong and we ended up we need to decode all possible combo I think we can revisit and the missing ones ?

Comment (jotak) [reply web/src/components/drawer/record/record-field.tsx:481]: Hey @msherif1234  .. sorry I missed these comments.
I didn't intend to say to write down the full list of possible combinations, but to replace that with a func that's able to decompose the bitfields into the list of atomic values. Just like I did here in excel : https://docs.google.com/spreadsheets/d/1CHfUQ8Sq0JyE_4ab-x1Aza8HobXXH8MmqI09Q0kLGkI/edit?gid=0#gid=0 :)
I'll open a follow-up

Currently, many aggregated flags aren't decomposed, e.g:
image

**Pull request merged**